src/snowflake/snowpark/functions.py

```diff
@@ -8551,7 +8551,13 @@ def iff(
 @publicapi
 def in_(
     cols: List[ColumnOrName],
-    *vals: Union["snowflake.snowpark.DataFrame", LiteralType, Iterable[LiteralType]],
+    *vals: Union[
+        "snowflake.snowpark.DataFrame",
+        LiteralType,
+        Column,
+        Iterable[LiteralType],
+        Iterable[Column],
```

Comment:
lets check with IR team if AST building needs to be updated. `build_expr_from_python_val(val_ast, val)` might fail for column type?

Reply:
sure, let me check with them